plugins/module_utils/network/sonic/config/route_maps/route_maps.py

```diff
@@ -0,0 +1,2340 @@
+                    if cmd_set_top.get('community') and 'community_attributes' in cmd_set_top['community']:
+                        cmd_community_attributes_set = set(cmd_set_top['community']['community_attributes'])
+                        command['set']['community'].pop('community_attributes')
+
+                    for cfg_community_attribute in cfg_community_attributes_set.difference(cmd_community_attributes_set):
+                        set_community_delete_attrs.append(self.set_community_rest_names[cfg_community_attribute])
+                        set_community_attributes_deletions.append(cfg_community_attribute)
+
+                    if set_community_attributes_deletions:
+                        # Update the list of deleted community attributes in the "command" dict.
+                        if not cmd_set_top.get('community'):
+                            command['set']['community'] = {}
+                        command['set']['community']['community_attributes'] = set_community_attributes_deletions
+
+                if command['set'].get('community') is not None and not command['set']['community']:
+                    command['set'].pop('community')
+
+                # Format and enqueue a request to delete eligible community attributes
+                if set_community_delete_attrs:
+                    bgp_set_delete_community_uri = bgp_set_delete_req_base + 'set-community'
+                    bgp_set_delete_comm_payload = \
+                        {'openconfig-bgp-policy:set-community': {}}
+                    bgp_set_delete_comm_payload_contents = \
+                        bgp_set_delete_comm_payload['openconfig-bgp-policy:set-community']
+                    bgp_set_delete_comm_payload_contents['config'] = \
+                        {'method': 'INLINE', 'options': 'REMOVE'}
+                    bgp_set_delete_comm_payload_contents['inline'] = \
+                        {'config': {'communities': set_community_delete_attrs}}
+
+                    request = {
+                        'path': bgp_set_delete_community_uri,
+                        'method': PATCH,
+                        'data': bgp_set_delete_comm_payload
+                    }
+                    dict_delete_requests.append(request)
+
+            # Check for deletion of set "extcommunity" lists. Delete the items in
+            # the currently configured list if it exists. As an optimization,
+            # avoid deleting list items that will be replaced by the received
+            # command.
+            set_extcommunity_delete_attrs = []
+
+            if 'extcommunity' not in cfg_set_top:
+                if command['set'].get('extcommunity'):
+                    command['set'].pop('extcommunity')
+                    if command['set'] is None:
+                        command.pop('set')
+                    return
+            else:
+                for extcomm_type in self.set_extcomm_rest_names:
```

Comment:
```suggestion
            else:
                command['set']['extcommunity'] = {}
                for extcomm_type in self.set_extcomm_rest_names:
```
Same as 'community'

Reply:
My response for this problem in handling of the "community" items applies here also. The same fix will take care of both cases.and also the "ipv6" case.